### PR TITLE
Run tests against react@next every night

### PR DESCRIPTION
**Pull request by tom-sherman**
Basically a re-do of #605 and #716

### CODE REVIEW

File: .github/workflows/react-next.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            node_modules
+            packages/*/node_modules
+          key: modules-${{ hashFiles('package-lock.json') }}
+
+      - name: Install dependencies
+        if: steps.cache-node-modules.outputs.cache-hit != 'true'
+        run: npm ci --no-audit
+
+      - name: Print version of react@next
+        run: npm view react@next && npm view react-dom@next
```
Comment (tom-sherman) [inline .github/workflows/react-next.yml:74]: This job is essentially the React 17 tests from the PR workflow with the addition of this step. Should aid in diagnosing bugs when they arise.

File: .github/workflows/react-next.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            node_modules
+            packages/*/node_modules
+          key: modules-${{ hashFiles('package-lock.json') }}
+
+      - name: Install dependencies
+        if: steps.cache-node-modules.outputs.cache-hit != 'true'
+        run: npm ci --no-audit
+
+  react-next:
+    name: Test React 17 (${{ matrix.shard }})
```
Comment (diegohaz) [inline .github/workflows/react-next.yml]:
```suggestion
    name: Test React Next (${{ matrix.shard }})
```

File: package.json [hunk 3]
```diff
@@ -4,11 +4,17 @@
   "scripts": {
     "start": "npm start -w website",
     "react17": "npm i --no-save react@17 react-dom@17 @testing-library/react@12",
+    "reactnext": "npm i --no-save react@next react-dom@next",
+    "reactnext": "npm i --no-save react@next react-dom@next",
```
Comment (diegohaz) [inline package.json]:
```suggestion
    "reactnext": "npm i --no-save react@next react-dom@next",
```

**Review by diegohaz — APPROVED** [review]
Thanks again @tom-sherman! 😀

**Pull request merged**